### PR TITLE
Password Hashing & Non-Working Tests Refactored

### PR DESCRIPTION
**Pull request by AndrewAchraf**
What I did:
- Refactored the user & admin login & signup to store a hashed version of the password in the DB.
- Added Lombok anntotations to UserSessionManager, UserService, and AdminService.
- Refactored UserService and AdminService Tests and added some more tests to them.

### CODE REVIEW

File: backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java [hunk 1]
```diff
@@ -31,35 +34,45 @@ void setUp() {
...
     void testLogin_InvalidPassword() {
         UserEntity mockUser = new UserEntity();
         mockUser.setEmail("test@example.com");
-        mockUser.setPassword("password123");
+        String rawPassword = "wrongpassword";
+        String encodedPassword = passwordEncoder.encode("password123");
+        mockUser.setPassword(encodedPassword);
 
         when(userRepository.findByEmail("test@example.com")).thenReturn(mockUser);
+        when(passwordEncoder.matches(rawPassword, encodedPassword)).thenReturn(false);
```
Comment (MohamedAbdo37) [inline backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java:58]: This line will invalidate the test, since the purpose of the test is to check that the method could find if raw Password matches with encoded Password.

File: backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java [hunk 2]
```diff
@@ -31,35 +34,45 @@ void setUp() {
     void testLogin_Successful() {
         UserEntity mockUser = new UserEntity();
         mockUser.setEmail("test@example.com");
-        mockUser.setPassword("password123");
+        String rawPassword = "password123";
+        String encodedPassword = passwordEncoder.encode(rawPassword);
+        mockUser.setPassword(encodedPassword);
 
         when(userRepository.findByEmail("test@example.com")).thenReturn(mockUser);
+        when(passwordEncoder.matches(rawPassword, encodedPassword)).thenReturn(true);
```
Comment (MohamedAbdo37) [inline backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java:42]: This line will invalidate the test, since the purpose of the test is to check that the method could find if raw Password matches with encoded Password.

File: backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java (hunk 1 above)
Comment (MohamedAbdo37) [inline backend/src/test/java/edu/alexu/cse/dripmeup/UserServiceTest.java:58]: the same problem

File: backend/src/test/java/edu/alexu/cse/dripmeup/AdminServiceTest.java [hunk 3]
```diff
@@ -1,83 +1,92 @@
...
-        String password = "password";
+        String rawPassword = "password";
+        String encodedPassword = passwordEncoder.encode(rawPassword);
         AdminEntity admin = new AdminEntity();
         admin.setUserName(userName);
-        admin.setPassword(password);
+        admin.setPassword(encodedPassword);
 
         when(adminRepository.findByUserName(userName)).thenReturn(admin);
+        when(passwordEncoder.matches(rawPassword, encodedPassword)).thenReturn(true);
```
Comment (MohamedAbdo37) [inline backend/src/test/java/edu/alexu/cse/dripmeup/AdminServiceTest.java:42]: > This line will invalidate the test, since the purpose of the test is to check that the method could find if raw Password matches with encoded Password. 

I have written this comment in UserService Test first time, so I mention it again.

**Review by MohamedAbdo37 — COMMENTED** [review]
good job @AndrewAchraf  you made great job. I have made some comments on test files.
Actually, I don't know if my notes is valid 100% which it's not clear the purpose of the new tests is it for test the login and signup with their corner cases or you test the encoding and decoding the password.


So , I will mark this review as Comment  not as Request changes till u clarify my queries.
Overall, great work, keep it up.

File: backend/src/main/java/edu/alexu/cse/dripmeup/service/UserService.java [hunk 4]
```diff
@@ -10,24 +12,25 @@
...
+        return user != null && passwordEncoder.matches(password, user.getPassword());
     }
 
     public Person signup(UserEntity user) throws HandlerException {
         if (userRepository.findByEmail(user.getEmail()) != null) {
             throw new HandlerException("Email already exists");
         }
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
         return new PersonDirector().construct(new UserPersonBuilder(user, userRepository));
     }
```
Comment (alihassann191) [inline backend/src/main/java/edu/alexu/cse/dripmeup/service/UserService.java:36]: In the **changePassword** method, the new password is directly saved without encoding it:
user.setPassword (newPassword.getPassword ()) ;
